### DISCUSSION TITLE
Alias columns comparison

### DISCUSSION THREAD

**Discussion by Khallil** [moj-analytical-services/splink#203] [Ideas]
Hi team, thanks for the great work with this project.

I'm trying to match an alias column for the name field, for example a company can have multiple names, legal ones, old ones, and in the case of cross sources matching, we have sources that have outdated names or in non-standard format, but still we want to be able to match them.

Did you thought about that scenario and is it implemented in the library yet ?

The alternative that I can do now is add the alias column in the `additional_columns_to_retain` field in the config and to do a custom comparaison like 
```
when name_l == name_r then 2
when name_l == alias_name_r then 2
```
in the name comparaison.

Have a great day

**Comment by RobinL** [comment]
Hi Khallil - thanks for the suggestion.

This is probably a lack of documentation (sorry) - I think it's possible to do this using what's called a 'custom comparison'.  

One of your comparison columns can be specified like this:
```
{
    "custom_name": "my_multi_column_comparison",
    "custom_columns_used": ["name_1", "name_alias"],
    "case_expression": "case when ..."",
"num_levels": 
}
```
as opposed to the more usual:

```
 {
            "col_name": "surname"
            "num_levels": 3,


        },
````
